src/api/mod.rs

```diff
@@ -403,6 +451,119 @@ impl<'a> ValueListBuilder<'a> {
     }
 }
 
+//TODO: move to its own module metadata, with related types
+fn to_meta_data(meta_hm: &HashMap<String, MetaValue>) -> Result<*mut meta_data_t, SubmitError> {
+    let meta;
+    unsafe {
+        meta = meta_data_create();
+    }
+    for (key, value) in meta_hm.iter() {
+        let c_key = CString::new(key.as_str()).map_err(|e| {
```

Comment:
If I'm not mistaken, I believe the collectd API immediately strdup's the input so this shouldn't need to be a `CString` (I believe `CStr` would be a better fix)